feat: heatmap year navi buttons

erispyu opened the pull request:
功能：当鼠标悬停在 heatmap 图标左侧或者右侧的时候，会显示向后或者向前的箭头，点击箭头可以切换到上一年或者下一年的数据。
TODO：我对JS和CSS不是很熟悉，按钮的风格和鼠标监听范围有待优化。

erispyu commented:
demo: https://erispyu.github.io/stats/

codepzj commented:
![image](https://github.com/user-attachments/assets/a868b063-a6f6-4f14-8bfa-90c71afe402b)
感谢pr，麻烦README不要改前面格式部分，revert一下，稍后同意

codepzj commented:
```javascript
var year = parseInt(heatmapChartDom.getAttribute('year')) || new Date().getFullYear();
const minYear = Math.min(...${data}.map(item => parseInt(item[0].split('-')[0])));
const maxYear = Math.max(...${data}.map(item => parseInt(item[0].split('-')[0])));

// 如果初始年份不在有效范围内，设置为最小年份
if (year < minYear || year > maxYear) {
    year = minYear;
}
```
轮播图可设置时间范围

erispyu committed:
feat: add heatmap year navigation buttons
- doc: update README.md
- chore: split large JavaScript file into modular components for better maintainability

erispyu commented:
@codepzj README.md 文件头更改已回退；轮播图已设置最小时间范围。

codepzj commented:
> @codepzj README.md 文件头更改已回退；轮播图已设置最小时间范围。

感谢大佬pr😀

xingwangzhe commented:
style:

![image](https://github.com/user-attachments/assets/2e8377c2-be06-4c01-b9f6-13d3eb6fc096)
暗色模式下，箭头不明显，可以改进一下
